### DISCUSSION TITLE
Is it possible to implement a OIDC pre-registration "hook" or redirect with access token?

### DISCUSSION THREAD

**Discussion by zikes** [ory/kratos#1264] [Q&A] [answered]
I have an OIDC provider that requires a second step to retrieve the user's profile information, [similar to Discord](https://github.com/ory/kratos/blob/2885146541f078be5488a44f2bec7d23afc021fd/selfservice/strategy/oidc/provider_discord.go#L70), and I'm wondering if it's possible to implement that behavior without coding it directly into Kratos. I'm not opposed to issuing a PR, but the provider situation is unusual in that it is a variation of Google OIDC, only it is querying the [YouTube v3 API](https://developers.google.com/youtube/v3/docs/channels/list) to retrieve their channel information.

Alternatively, if there is a way to retrieve the session's access token after login is completed I could have my application perform that query itself.

**Comment by aeneasr** [comment, marked as answer]
There’s a feature request (https://github.com/ory/kratos/issues/397) that’s not implemented yet but I think thatvwouldbbe the best option!
